### PR TITLE
`Programming exercises`: Add removal of duplicate test cases assigned to multiple tasks

### LINKED ISSUE #7670: Wrongly displayed grading table when a test case is linked in two different tasks
### Describe the bug

You can include the test cases in the individual subtasks of programming assignments so that students can click directly on the feedback for the subtask.
The grading page also shows it sorted according to the subtasks.

Problem: If you accidentally link a test in two subtasks, it also shows the grading table in both subtasks, and then, for example, the "Resulting Points" are no longer correct.

Suggestion to solve this problem: Either warn about this in the task editor and/or, if the tests have been linked twice, display all tests individually in the grading table and don't sort by subtasks.

### To Reproduce

1. Create a programming exercise
2. Include different test cases to the subtasks
3. Link a test case to two different subtasks
4. Go to the grading table
5. The grading table displays that test case twice and shows wrong information, e.g. the Resulting points are off

### Expected behavior

To not see both subtasks in the grading table and correct "Resulting Points".

### Screenshots

image

### Which version of Artemis are you seeing the problem on?

6.6.8

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by simon-hng**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Resolves #7670

### Description

In the task-based grading view, removes test cases that are assigned to multiple tasks.
The task the test case is shown in ultimately is the first task it is found in. (this ordering comes from the server and is arbitrary)

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Create a programming exercise
1. Include different test cases to the subtasks
1. Link a test case to two different subtasks
1. Go to the grading table
1. The grading table displays that test case once and shows correct grading information

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| [app/exercises/programming/manage/grading/tasks/programming-exercise-task](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5793-7/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task/index.html) | 93.69% | ✅                           |
| [app/exercises/programming/manage/grading/tasks/programming-exercise-task](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5793-7/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task/index.html) | 92.3% | ✅              |

### Screenshots

Test case `testMergeSort` is included in both `Implement Bubble Sort` and `Implement Merge Sort`
image

The test case is only attributed to implement bubble sort (because this tasks appeared first in the ordering given by the server)

The task `Implement Merge Sort` is removed since it does not contain any test cases and no viable grading can be configured for such a task
image

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-task.service.ts [hunk 1]
```diff
@@ -204,18 +230,9 @@ export class ProgrammingExerciseTaskService {
         return task;
     };
 
-    private configureTestCases = (task: ProgrammingExerciseTask): ProgrammingExerciseTask => {
```
Comment (simon-hng) [inline src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-task.service.ts:207]: This is no longer needed, as duplicate test cases are not allowed

**Review by JanaNF — DISMISSED** [review]
Manual tested on ts2. Worked as expected.

**Review by sarpsahinalp — DISMISSED** [review]
Tested on ts2. Worked as expected.

**Review by LenaKahle — APPROVED** [review]
Tested on legacy ts11, works as described.

**Review by AntonGeTUM — COMMENTED** [review]
Tested on TS11: When a programming exercise is created first, and the duplicate test case is added afterward, both tasks remain visible in "Grading". However, it does remove the duplicate test case.
Other than that, it works as expected.

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Review by vinceclifford — APPROVED** [review]
Works as expected, tested on TS2

**Review by Predixx — DISMISSED** [review]
Tested on ts2 with Vincent Clifford. Looks like the screenshots

**Comment by krusche** [comment]
Client tests are missing

**Comment by simon-hng** [comment]
Thanks @Strohgelaender !

**Review by Strohgelaender — APPROVED** [review]

**Review by JanaNF — APPROVED** [review]
Re-approve

**Review by LenaKahle — APPROVED** [review]
Reapprove.

**Review by florian-glombik — APPROVED** [review]
Approve code - clean and good to understand!

**Pull request merged**